### ISSUE TITLE
Auth error with Chrome extension: Invalid Cookie Policy

### ISSUE THREAD

**Issue by yaraju** [GoogleWebComponents/google-signin#102]
When using google-sign-in on Chrome extension, loading the page directly like so:

```
chrome-extension://mylongid/my-element/demo.html
```

With the following tag in my custom element:
```xml
<google-signin client-id="myextensionclientId" scopes="https://www.googleapis.com/auth/userinfo.email" width="iconOnly" brand="short"></google-signin>
```
I get the following error::

**Error:**
*Uncaught gapi.auth2.ExternallyVisibleError: Invalid cookiePolicy*
```
tY	@	cb=gapi.loaded_0:120
_.hN	@	cb=gapi.loaded_0:133
AuthEngine.initAuth2	@	google-signin-aware.html.0.js:139
(anonymous function)	@	api.js?onload=https___apis_google_com_js_api_js_onload___callback___api_loaded:7
ga	@	api.js?onload=https___apis_google_com_js_api_js_onload___callback___api_loaded:1
v.(anonymous function).b	@	api.js?onload=https___apis_google_com_js_api_js_onload___callback___api_loaded:7
v.(anonymous function)	@	api.js?onload=https___apis_google_com_js_api_js_onload___callback___api_loaded:7
I.(anonymous function)	@	api.js?onload=https___apis_google_com_js_api_js_onload___callback___api_loaded:7
(anonymous function)	@	cb=gapi.loaded_0:1
```

**Comment by yaraju** [comment]
Subsequently a red icon with no "g" shows up. On Clicking the red button, I get the following error:

```
Uncaught TypeError: Cannot read property 'currentUser' of nullAuthEngine.signIn @ google-signin-aware.html.0.js:326Polymer.signIn @ google-signin-aware.html.0.js:493Polymer.signIn @ google-signin.html.0.js:407handler @ polymer.html.0.js:361decorated @ polymer.html.0.js:3119
```

**Comment by ebidel** [comment]
My suspicion is that this is an issue with the Google Signin API not working from `chrome-extension://` origin. cc @Zoramite

**Comment by Scarygami** [comment]
Yes, `gapi` generally doesn't get along well with Chrome extensions, see for example: https://github.com/google/google-api-javascript-client/issues/64

The preferred way for authentication inside of Chrome extensions is the [Chrome Identity API](https://developer.chrome.com/apps/identity) which would probably be out-of-scope for this element.

**Comment by yaraju** [comment]
Hmm. I suppose an alternative would be to somehow proxy these calls through the background script, so that the element just gets the status.

But this would probably fall within the scope of a separate set of extensions that work on Chrome Apps/Extensions, and not within the core GoogleWebComponents? Is there a more appropriate place where I can put in or track a request for such a set of components from the Polymer team? (Or would that be the Chrome team?)

**Comment by addyosmani** [comment]
I think the Chrome Identity API wouldn't be the best fit here - agree on it being out of scope. The element should focus on the 90% use-case. We could (potentially) document some of the workarounds needed if operating in an extensions world, but imo my take is we should avoid baking in workarounds here into the element itself.

**Comment by JeffPinkston** [comment]
I'm getting the exact same errors during build and onClick() when I run my Polymer project on my device using Cordova. Is that related? The code works as expected when I run locally on my desktop.

**Comment by atotic** [comment]
Could you use oauth2 the way it is intended for installed applications in this use case?

https://developers.google.com/identity/protocols/OAuth2InstalledApp

That would be a nice addition to google-signin, it is not an uncommon request, and we could hide the differences under the hood.

**Comment by neaumusic** [comment]
Sign-In to get OpenID id_token from a chrome extension seems impossible without exposing it in the URL and whitelisting / routing to https:// protocol like deprecated https://asdfmychromextensionshaasdf.chromiumapp.org/

This could have been possible with `chrome.windows.create` incognito, but there is no way to whitelist the chrome extension as a certified endpoint, which kinda makes sense(?)

Ultimately, since the cookie policy is invalid and a chrome extension has no private keys, it's impossible to know that my app is the one who's sending the id_token to my backend. It could have been intercepted by any tab change listening extensions.

My only workaround seems that I'd have to host my front-end code on my own server, rather than using the chrome extension APIs. I don't even think I can use an iframe inside my browser popup that accesses my app, since again I have no way to validate my extension is the one listening for `window.postMessage` or whatever.

Not sure this makes sense, but ultimately I think the chrome webstore needs to provide some server authentication stuff (and support OpenID) rather than just giving the identity hook for paid gapi services.

**Comment by srikanta72** [comment]
I am getting the same error.
<html>
<head>
    <style>
        /* Put your CSS over here */
        
     </style>
     
    <script src="https://apis.google.com/js/api.js"></script>
    <script
  src="https://code.jquery.com/jquery-3.4.1.js"
  integrity="sha256-WpOohJOqMqqyKL9FccASB9O0KwACQJpFTUBLTYOVvVU="
  crossorigin="anonymous"></script>
    <script>
        function start() {
            // 2. Initialize the JavaScript client library.
            gapi.client.init({
                'apiKey': 'MY_API_KEY',
                // clientId and scope are optional if auth is not required.
                'clientId': 'MY_CLIENT_ID.apps.googleusercontent.com',
                'scope': 'profile',
            }).then(function() {
                // 3. Initialize and make the API request.
                return gapi.client.request({
                    'path': 'https://people.googleapis.com/v1/people/me?requestMask.includeField=person.names',
                })
            }).then(function(response) {
                console.log(response.result);
            }, function(reason) {
                console.log('Error: ' + reason.result.error.message);
            });
        };
        // 1. Load the JavaScript client library.
        gapi.load('client', start);

        $(".signin-btn").on("click", function(){
  alert("The paragraph was clicked.");
});
        
    </script>

    
</head>

<body>
    <div id="signin">

        <input type="button" class="sign-in-btn" value="Sign in with Google" />
    </div>
</body>

</html>

**Comment by harleenarora** [comment]
how to solved this issue, on chrome extension.
